token/cli/src/main.rs

```diff
@@ -943,11 +953,34 @@ fn command_burn(
     })
 }
 
+
+fn check_if_amount_overflows_supply(  
+    config: &Config,
+    token: Pubkey,
+    ui_amount : u64,
+) -> bool {
+    let supply = config.rpc_client.get_token_supply(&token).unwrap();
+    let amount = ui_amount;
+    if amount > u64::MAX {
+        return true;
+    };
+    let supply_val = supply.amount;
+    let supply_u64: u64 = supply_val.parse().unwrap();
```

Comment:
nit: can condense these into
```suggestion
    let supply: u64 = supply.amount.parse().unwrap();
```